data/map_data.s

```diff
@@ -1,5 +1,355 @@
+.include "constants/constants.inc"
+.include "asm/macros.inc"
+
 	.section .rodata
- 
-    .global gUnknown_086EA408
-gUnknown_086EA408:
-    .incbin "baserom.gba", 0x006EA408, 0x3E3870
+
+@ TODO: Include as much data in C files as possible
+
+@ NOTE: Sometimes in the codebase, there is pointers to inside the tileset data.
+@       Set these pointers to:
+@           (<Beginning_of_Tileset> + (tile_count * TILE_SIZE_4BPP))
+
+
+@ NOTE: For some reason each level (except for the dummies) has a different tileset, even those that are visually identical.
+@       Palettes are also stored as copies of each level.
+@       We will just include the same file in those instances, unless there is actually a difference of course.
+@       It is something to keep in mind in case one wants to port the game to other platforms down the line.
+
+@
+@  --- Leaf Forest ---
+@
+
+@
+@ --- Leaf Forest - Act 1 ---
+@
+
+    .global Palette_Stage_LeafForest_Act1
+Palette_Stage_LeafForest_Act1: @ 0x086E9E08
+    .incbin "graphics/tilesets/zones/1_leaf_forest/tileset.gbapal"
+    mDataSize Palette_Stage_LeafForest_Act1
+
+    .global Tileset_Stage_LeafForest_Act1
+Tileset_Stage_LeafForest_Act1: @ 0x086EA008
+    .incbin "graphics/tilesets/zones/1_leaf_forest/tileset_act1.4bpp"
+    mDataSize Tileset_Stage_LeafForest_Act1
+
+    .global gUnknown_086F12A8
+gUnknown_086F12A8: @ 0x086F12A8
+    .incbin "data/maps/zone1/act1/gUnknown_086F12A8.raw"
```

Comment:
For now should name these files `.bin`

Reply:
I can't cause that would lead them to be RL-compressed, which we don't want.
And I think it's not a bad idea to differentiate between data that'll go into the ROM uncompressed vs. compressed.